[Kou Yong Kang] iP

kouyk opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

kouyk committed:
Duke.java: implement skeletal version

kouyk committed:
Duke.java: store the commands given

kouyk committed:
Add Task.java

kouyk committed:
Duke.java: use task and enable marking as done

kouyk committed:
Override toString of task to refactor code

kouyk committed:
Create Todo, Deadline and Event task types, refactored code to enable

kouyk committed:
Setup automated tests using sample scripts

kouyk committed:
Deal with whitespace properly and implement exceptions to catch invalid input

kouyk committed:
Duke.java: Add support for deleting tasks from the list

kouyk committed:
test: check for delete task

kouyk committed:
Use enums for functions

kouyk committed:
Refactor printing of lines

kouyk committed:
Update .gitignore

kouyk committed:
Implement task export as a one-liner

kouyk committed:
Add TaskSaver.java for save and restore functionality

kouyk committed:
Save and restore tasks from tasks.txt

kouyk committed:
Update .gitignore: ignore tasks.txt

kouyk committed:
TaskSaver: Minor refactoring for better style

kouyk committed:
Refactor task creation

kouyk committed:
Remove redundant processDescription

kouyk committed:
Parse date and time string for events and deadlines

kouyk committed:
Extend list command to accept a date to list relevant tasks

kouyk committed:
Merge branch 'branch-Level-7'

Implement save and restore functions

kouyk committed:
Merge branch 'branch-Level-8'

kouyk committed:
Minor cleanup and renaming

kouyk committed:
Rename TaskSaver to Storage

kouyk committed:
Introduce TaskList for better encapsulation

kouyk committed:
Storage: accept TaskList when saving

kouyk committed:
Refactor for more OOP

kouyk committed:
Package everything

kouyk committed:
More abstraction for tasks with date and time

When exporting tasks with date and time, relegate the task of appending
datetime string to the super class to avoid repeated code.

kouyk committed:
Add unit tests for all task classes

kouyk committed:
Update .gitignore

kouyk committed:
Fix file not saving when directory exists

kouyk committed:
Add javadocs

kouyk committed:
Adjust code style

kouyk committed:
Add find command

The find command enables the user to input a keyword and get a list of
tasks with descriptions that contain the specified keyword.

kouyk committed:
Merge branch 'branch-A-JavaDoc'

kouyk committed:
Merge branch 'branch-A-CodingStandard'

Update all files to conform with the coding standard

kouyk committed:
Merge branch 'branch-Level-9'

kouyk committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

kouyk committed:
Configure checkstyle

kouyk committed:
Configure gradle for ekud

kouyk committed:
Minor renaming and module reorganisation

kouyk committed:
Fix style violations

kouyk committed:
Include version in jar naming

kouyk committed:
Fix more build warnings

kouyk committed:
Update README.md

kouyk committed:
Correct name of Ekud

kouyk committed:
Include javafx in dependencies

kouyk committed:
Set arrayInitIndent to 8 in checkstyle.xml

kouyk committed:
Implement GUI using JavaFX

kouyk committed:
Tweak indentation and status indicators

GUI does not use longer fixed-width characters, hence indentation needs
to be adjusted accordingly. Since the status icon function has been
implemented since the beginning, utilise that function too and replace
the X used in the text-based UI.

kouyk committed:
Remove text-based UI

kouyk committed:
Add JavaDoc for getResponse function

WeiLiangLOL commented on src/main/java/ekud/ui/DialogueBox.java:
```diff
@@ -0,0 +1,75 @@
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
+
+public class DialogueBox extends HBox {
```
It'd be nice to leave a Javadoc header for the DialogueBox class!

WeiLiangLOL commented on src/main/java/ekud/task/TaskList.java:
```diff
@@ -0,0 +1,154 @@
+        taskArray.add(newTask);
+    }
+
+    /**
+     * Remove a task by index.
+     *
+     * @param index The index of the task to be removed.
+     * @return Task removed from list.
+     * @throws InvalidTaskIndexException If the index is out of range (index < 0 || index >= size()).
```
I like that the error thrown is very self explanatory!

pPris replied:
Agreed with this reviewer, it's easy to understand what errors are being expected especially with your javadoc comment too

WeiLiangLOL commented on src/main/java/ekud/task/Task.java:
```diff
@@ -0,0 +1,66 @@
+
+    public String getDescription() {
+        return description;
+    }
```
Some of the methods, though already very self descriptive, could use some javadoc comments as well.

WeiLiangLOL commented on src/main/java/ekud/MainApp.java:
```diff
@@ -0,0 +1,18 @@
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+
+public class MainApp extends Application {
```
Could use some javadocs!

WeiLiangLOL commented:
I like your code overall, it's very clean 👍

pPris commented on src/main/java/ekud/command/ListCommand.java:
```diff
@@ -0,0 +1,76 @@
+ * Command that lists all tasks, optionally filter by date if given.
+ */
+public class ListCommand extends Command {
+    private final boolean filter;
```
Could this variable be named to sound more 'boolean'? (I did like that you've followed boolean naming conventions in other places)

pPris commented on src/test/java/ekud/task/DeadlineTest.java:
```diff
@@ -0,0 +1,68 @@
+
+    @Nested
+    @DisplayName("freshly created")
+    class FreshTask {
```
Perhaps there could be a brief comment on what a fresh task means here?

pPris reviewed (commented):
Overall I found your code really clean, it seems to adhere to the coding standard really well.

kouyk committed:
Ekud: handle to bye commands

kouyk committed:
Terminate programme after 3 seconds once bye command is issued

kouyk committed:
Update checkstyle.xml to check for missing Javadoc for class definitions

kouyk committed:
Docs: Add missing Javadocs for class definitions

kouyk committed:
command: minor renaming to follow code style

kouyk committed:
Exit the application through another thread

As JavaFX is single-threaded, no GUI updates is possible if event
handler does not return. Hence the workaround is to start a thread that
counts down and invokes the exit function.

kouyk committed:
Use more assertions

kouyk committed:
Extract exception messages into a single class

kouyk committed:
Minor refactoring for better code quality

kouyk committed:
Use lambda for execution of find command

kouyk committed:
Expose underlying container of TaskList throught its Stream

kouyk committed:
Use stream in find and list commands

kouyk committed:
Merge pull request #1 from kouyk/branch-A-Assertions

Use more assertions

kouyk committed:
Merge branch 'master' into branch-A-CodeQuality

kouyk committed:
Merge branch 'master' into branch-A-Lambdas

kouyk committed:
Merge branch 'master' into branch-A-Streams

kouyk committed:
Merge pull request #2 from kouyk/branch-A-CodeQuality

Improve code quality

kouyk committed:
Merge branch 'master' into branch-A-Lambdas

kouyk committed:
Merge branch 'master' into branch-A-Streams

kouyk committed:
Merge pull request #3 from kouyk/branch-A-Lambdas

Use lambda for execution of find command

kouyk committed:
Merge branch 'master' into branch-A-Streams

kouyk committed:
Merge pull request #4 from kouyk/branch-A-Streams

Use streams

kouyk committed:
Release v0.2

kouyk committed:
Show welcome message on start up

kouyk committed:
Move the scheduling viewing functionality into a new view command

kouyk committed:
Setup initial github actions

kouyk committed:
Update tests to check for icons instead of X

kouyk committed:
Fix tests for Windows

Windows handles emoji differently, tests should continue to use the
actual unicode rather than the copied emoji.

kouyk committed:
Update README.md

kouyk committed:
Prevent the label truncating long text

When the total height of all the DialogueBoxes exceed the display
height, the height of every DialogueBox is adjusted to compensate for
it, until a minimum of 2 lines are shown. This is not desired since it
will mean that text is truncated which does not make sense for a chat
dialogue.

This is resolved by setting the minHeight of the DialogueBox to
USE_PREF_HEIGHT, and therefore the DialogueBox will maintain the correct
height.

kouyk committed:
Switch to using different image for both bot and user

kouyk committed:
Update README.md to include acknowledgements for media usage

kouyk committed:
Use a different dialogue box for Ekud

kouyk committed:
Add Ekud's profile picture

kouyk committed:
Set window title and prevent resizing of window

kouyk committed:
Tweak GUI colours and layout

kouyk committed:
Fix spacing and indentation for view command

kouyk committed:
Improve code quality

kouyk committed:
Add screenshot of application

kouyk committed:
Update user guide

kouyk committed:
Release v0.3

kouyk committed:
More Javadoc

kouyk committed:
Use jar file directory as working directory if possible

kouyk committed:
Lower height of window
